### PR TITLE
Add some notes about breaking changes

### PR DESCRIPTION
**Pull request by dinosaure**

### CODE REVIEW

**Review by mseri — APPROVED** [review]

**Review by mseri — COMMENTED** [review]
I have fixed a couple of minor typos and slightly changed the wording of the last sentence, I hope you don't mind

**Comment by mseri** [comment]
I am going to report this also on cohttp-3.0.0 PR

File: CHANGES.md [hunk 1]
```diff
@@ -8,6 +8,50 @@
...
+
+  [Release of Conduit 3.0.0](https://github.com/mirage/ocaml-conduit/releases/tag/v3.0.0)
+  [HOW-TO use Conduit 3.0.0](https://mirage.github.io/ocaml-conduit/conduit/howto.html)
+  [Documentation of Conduit 3.0.0](https://mirage.github.io/ocaml-conduit/conduit/Conduit/module-type-S/index.html)
+
+  About the server-side part of Cohttp, we broke the API according to Conduit 3.0.0. The user
+  must update the way to launch a Cohttp server. The technical update is about the `~mode`
+  argument which disappears to let 3 arguments: `cfg`, `service` and `protocol`.
+
+  They permit to the user to choose which kind of service he wants (a TLS service or
```
Comment (samoht) [inline CHANGES.md:25]: "This allow users to choose which kind of service they want"

File: CHANGES.md [hunk 2]
```diff
@@ -8,6 +8,50 @@
...
+  `service` value. For example, for `Conduit_lwt.TCP.service`, `cfg = Conduit_lwt.TCP.configuration`.
+  On this way, the user is able to launch a HTTP server with:
+
+```ocaml
+let cfg =
+  { Conduit_lwt.TCP.sockaddr= Unix.ADDR_INET (Unix.inet_addr_loopback, 8080)
+  ; capacity= 40 }
+
+let run cohttp_config =
+  Cohttp_lwt_unix.Server.create cfg Conduit_lwt.TCP.protocol Conduit_lwt.TCP.service
```
Comment (samoht) [inline CHANGES.md:42]: out of curiosity when do weed need to both specify a protocol and a service? I though the protocol was included in the service?

Comment (dinosaure) [reply CHANGES.md:42]: It's needed to explain how to _pack_ the given flow. The current implementation of `Conduit.serve_with_handler` gives the real type/value. Then, `Cohttp` packs this value to the abstract `Conduit.flow` value (but it needs to know the _protocol_ to do this operation).

Comment (samoht) [reply CHANGES.md:42]: that part of the API is still a bit confusing to me ; I'm wondering if we could make this simpler by having to somehow pass only the service (for instance by extending a bit the definition of what a service is).

Comment (samoht) [reply CHANGES.md:42]: (note: I am fine with the current API, I'm just suggesting possible improvement for conduit 3.1)

Comment (mseri) [reply CHANGES.md:42]: We still have time to change it if there is an agreement, and avoid another breaking ( I would call it 4.0.0) release :P

Comment (avsm) [reply CHANGES.md:42]: Agreed, I find this bit confusing too. Now's the time to tweak it, not 3.1 :-)

Comment (mseri) [reply CHANGES.md:42]: Lol I had just merged the PR. I can remove the announcement

**Pull request merged**